src/v2/spec.yaml

```diff
@@ -513,100 +514,221 @@ components:
-        updatedTime:
+          description: The url of the resource provider.
+        cache_age:
+          oneOf:
+            - type: string
+            - type: number
+          description: The suggested cache lifetime for this resource, in seconds. Consumers may choose to use this value or not.
+        thumbnail_url:
           type: string
+          description: A URL to a thumbnail image representing the resource. The thumbnail must respect any maxwidth and maxheight parameters. If this parameter is present, thumbnail_width and thumbnail_height must also be present.
+        thumbnail_width:
+          type: number
+          description: The width of the optional thumbnail. If this parameter is present, thumbnail_url and thumbnail_height must also be present.
+        thumbnail_height:
+          type: number
+          description: The height of the optional thumbnail. If this parameter is present, thumbnail_url and thumbnail_width must also be present.
+
+    LinkTypeData:
+      allOf:
+        - $ref: '#/components/schemas/OembedData'
+        - type: object
+          properties:
+            type:
+              type: string
+              enum: [link]
+              readOnly: true
+
+    PhotoTypeData:
+      allOf:
+        - $ref: '#/components/schemas/OembedData'
+        - type: object
+          required:
+            - url
+            - width
+            - height
+          properties:
+            type:
+              type: string
+              enum: [photo]
+              readOnly: true
+            url:
+              type: string
+              description: The source URL of the image. Consumers should be able to insert this URL into an <img> element. Only HTTP and HTTPS URLs are valid.
+            width:
+              type: number
+              description: The width in pixels of the image specified in the url parameter.
+            height:
+              type: number
+              description: The height in pixels of the image specified in the url parameter.
+
+    VideoTypeData:
+      allOf:
+        - $ref: '#/components/schemas/OembedData'
+        - type: object
+          required:
+            - html
+            - width
+            - height
+          properties:
+            type:
+              type: string
+              enum: [video]
+              readOnly: true
+            html:
+              type: string
+              description: The HTML required to embed a video player. The HTML should have no padding or margins. Consumers may wish to load the HTML in an off-domain iframe to avoid XSS vulnerabilities.
+            width:
+              type: number
+              description: The width in pixels required to display the HTML.
+            height:
+              type: number
+              description: The height in pixels required to display the HTML.
+
+    RichTypeData:
+      allOf:
+        - $ref: '#/components/schemas/OembedData'
+        - type: object
+          required:
```

Comment:
Do we need to mark these fields as required?

e.g. if a page doesn't include `html` or another field, do we skip indexing the oembed? if we do index it, and if any of these fields is `null`, do we skip returning the object at the API layer?

The example here has height as `null` https://github.com/neynarxyz/nindexer/pull/65

Reply:
These new types just mirror what the library's original Typescript types do https://github.com/extractus/oembed-extractor/blob/e3b9aff4084118e12b6e98bdf997a78901ea6dfd/index.d.ts#L30-L107. So the library is saying what it returns to us will always have these three keys. Key being required doesn't mean the value has to be non-null. 

But after looking closer, this does mean the values of these keys should be marked as nullable, which they aren't. Will fix.

Reply:
Fixed